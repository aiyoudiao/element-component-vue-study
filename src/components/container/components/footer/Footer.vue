<script setup lang="ts">
import { withDefaults } from 'vue'
interface FooterType {
  height?: string
}

withDefaults(defineProps<FooterType>(), {
  height: '60px',
})
</script>

<script lang="ts">
export default {
  name: 'ElFooter',
}
</script>

<template>
  <footer class="el-footer" :style="{ height }">
    <slot />
  </footer>
</template>

<style lang="scss">
@import '../../../../theme/src/footer.scss';
</style>
