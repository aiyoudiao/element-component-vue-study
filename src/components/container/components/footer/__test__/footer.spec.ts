import { mount } from '@vue/test-utils'
import Footer from '../Footer.vue'
describe('Footer.vue', () => {
  it('Footer snapshot', () => {
    const wrapper = mount(Footer)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('Footer render', () => {
    const wrapper = mount(Footer)
    expect(wrapper.classes()).toContain('el-footer')
  })
  it('Footer slot', () => {
    const wrapper = mount(Footer, {
      slots: {
        default: () => 'hello world',
      },
    })
    expect(wrapper.text()).toBe('hello world')
  })
  it('Footer style', () => {
    const wrapper = mount(Footer, {
      props: {
        height: '90px',
      },
    })
    const height = wrapper.vm.$el.style.height
    expect(height).toBe('90px')
  })
})

