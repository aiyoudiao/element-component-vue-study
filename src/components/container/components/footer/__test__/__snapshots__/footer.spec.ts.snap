// Vitest Snapshot v1

exports[`Footer.vue > Footer snapshot 1`] = `
<footer
  class="el-footer"
  style="height: 60px;"
>
  
  
</footer>
`;
